caikit/runtime/servicers/info_servicer.py

```diff
@@ -0,0 +1,84 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the implementation for retrieving information about the
+library and services.
+"""
+# Have pylint ignore Class XXXX has no YYYY member so that we can use gRPC enums.
+# pylint: disable=E1101
+
+# Standard
+from typing import Dict
+
+# Third Party
+import importlib_metadata
+
+# First Party
+import alog
+
+# Local
+from caikit.config import get_config
+from caikit.interfaces.runtime.data_model import RuntimeInfoResponse
+
+log = alog.use_channel("RI-SERVICR-I")
+
+
+class InfoServicer:
+    """This class contains the implementation for retrieving information about the
+    library and services."""
+
+    def GetRuntimeInfo(
+        self, request, context  # pylint: disable=unused-argument
+    ) -> RuntimeInfoResponse:
+        """Get information on versions of libraries and server for GRPC"""
+        return self.get_runtime_info_impl()
+
+    def get_runtime_info_impl(self) -> RuntimeInfoResponse:
+        """Get information on versions of libraries and server from config"""
+        versions = {}
+        version_info = get_config().runtime.version_info or {}
+        if version_info.get("python_packages"):
+            all_packages = version_info.get("python_packages").get("all_packages")
```

Comment:
This is actually unsafe since `all_packages` may not be defined below

```suggestion
        all_packages = (version_info.get("python_packages") or {}).get("all_packages")
```